plutus-ledger-api/exe/test-onchain-evaluation/Main.hs

```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE LambdaCase    #-}
+{-# LANGUAGE TupleSections #-}
+
+module Main (main) where
+
+import LoadScriptEvents (eventsOf, loadEvents)
+
+import PlutusCore.Pretty
+import PlutusLedgerApi.Common
+import PlutusLedgerApi.Test.EvaluationEvent
+import PlutusLedgerApi.V1 qualified as V1
+import PlutusLedgerApi.V2 qualified as V2
+
+import Control.Concurrent.Async (mapConcurrently)
+import Control.Exception (evaluate)
+import Control.Monad.Extra (whenJust)
+import Control.Monad.Writer.Strict
+import Data.List.NonEmpty (nonEmpty)
+import Data.Maybe (catMaybes)
+import System.Directory.Extra (listFiles)
+import System.Environment (getEnv)
+import System.FilePath (isExtensionOf, takeBaseName)
+import Test.Tasty
+import Test.Tasty.HUnit
+
+-- | Test cases from a single event dump file
+testOneFile :: FilePath -> TestTree
+testOneFile eventFile = testCase (takeBaseName eventFile) $ do
+    events <- loadEvents eventFile
+    case ( mkContext V1.mkEvaluationContext (eventsCostParamsV1 events)
+         , mkContext V2.mkEvaluationContext (eventsCostParamsV2 events)
+         ) of
+        (Right ctxV1, Right ctxV2) -> do
+            errs <-
+                fmap catMaybes $
+                    mapConcurrently
```

Comment:
That was already in this code (I copied this and modified it to do the analysis and then did some refactoring to share some common code).  I wasn't sure about `mapConcurrently` so I just left it.

Reply:
That's @zliu41 's code, so I didn't want to interfere with it (it's changed in this  PR because I did a bit of refactoring to factor out some shared code).